### PR TITLE
Applied comments in PR 822

### PR DESCRIPTION
**Pull request by lziq**
Some comments original from https://github.com/partiql/partiql-lang-kotlin/pull/822 are for `main` branch instead of OTS feature branch. This PR aims to apply those comments to `main`. The changes include: 
1. Make function `PartiqlPhysical.Type.toTypedOpParameter()` internal
2. Remove arguments from `PartiqlAst.Type.toTypedOpParameter()` and make it private 
3. Make `PartiqlAstSanityValidator` internal. 
4. Removed `PartiqlPhysicalSanityValidator`, and moved the validation there to `AstToLogicalVisitorTransform`

Note that don't worry too much about if passing argument of `TypedOpParameters` is correct, since we will deprecate and remove it as we approach PartiQL 1.0

### CODE REVIEW

File: lang/src/org/partiql/lang/planner/transforms/AstToLogicalVisitorTransform.kt [hunk 1]
```diff
@@ -326,6 +365,55 @@ internal class AstToLogicalVisitorTransform(
...
+            if (setQuantifier is PartiqlAst.SetQuantifier.Distinct && metas.containsKey(IsCountStarMeta.TAG)) {
+                err(
+                    "COUNT(DISTINCT *) is not supported",
+                    ErrorCode.EVALUATOR_COUNT_DISTINCT_STAR,
+                    errorContextFrom(metas),
+                    internal = false
+                )
+            }
+        }
 }
```
Comment (RCHowell) [inline lang/src/org/partiql/lang/planner/transforms/AstToLogicalVisitorTransform.kt:417]: Validation should be its own pass using Visitor. It's not a transform

File: lang/src/org/partiql/lang/planner/transforms/LogicalToLogicalResolvedVisitorTransform.kt [hunk 2]
```diff
@@ -195,6 +198,22 @@ internal data class LogicalToLogicalResolvedVisitorTransform(
...
+            locals.forEachIndexed { idx, it ->
+                if (it.registerIndex.value != idx.toLong()) {
+                    throw EvaluationException(
+                        message = "Variable index must match ordinal position of variable",
+                        errorCode = ErrorCode.INTERNAL_ERROR,
+                        errorContext = propertyValueMapOf(),
+                        internal = true
+                    )
+                }
+            }
```
Comment (RCHowell) [inline lang/src/org/partiql/lang/planner/transforms/LogicalToLogicalResolvedVisitorTransform.kt:216]: again, validation doesn't belong in transform passes. Were these changes meant to be included?

File: lang/src/org/partiql/lang/planner/transforms/AstToLogicalVisitorTransform.kt (hunk 1 above)
Comment (lziq) [reply lang/src/org/partiql/lang/planner/transforms/AstToLogicalVisitorTransform.kt:417]: Will apply this comment

File: lang/src/org/partiql/lang/planner/transforms/LogicalToLogicalResolvedVisitorTransform.kt (hunk 2 above)
Comment (lziq) [reply lang/src/org/partiql/lang/planner/transforms/LogicalToLogicalResolvedVisitorTransform.kt:216]: I agree that validation should have its own pass. But do you think we really need the check here?

Comment (lziq) [reply lang/src/org/partiql/lang/planner/transforms/LogicalToLogicalResolvedVisitorTransform.kt:216]: It is trying to ensure the code we are writing is correct

**Review by RCHowell — APPROVED** [review]

**Pull request merged**